storage/remote/queue_manager_test.go

```diff
@@ -65,89 +66,80 @@ func newHighestTimestampMetric() *maxTimestamp {
+				{behaviour: errors.New("pretend unrecoverable err"), attemptString: "0,pretend unrecoverable err"},
 			},
 		},
-		// Now test where the server suddenly stops speaking 2.0 and we need to downgrade because it returns a 400.
 		{
-			success: true, name: "v2 request to v2 server that has downgraded via 400", qmRwFormat: config.RemoteWriteProtoMsgV2, rwFormat: Version2, steps: []contentNegotiationStep{
-				{lastRWHeader: "2.0;snappy,0.1.0", compression: "snappy", behaviour: &ErrRenegotiate{"", 400}, attemptString: "0,1,snappy,HTTP 400: msg: "},
-				{lastRWHeader: "0.1.0", compression: "snappy", behaviour: nil, attemptString: "0,0,snappy,ok"},
+			success: false, name: "v2 talks to v1 that tries to unmarshal v2 payload with v1 proto", senderProtoMsg: config.RemoteWriteProtoMsgV2, receiverProtoMsg: config.RemoteWriteProtoMsgV1, steps: []contentNegotiationStep{
+				{behaviour: nil, attemptString: "0,ok"}, // It's ok, because payload can be unmarshaled just fine, but it's empty.
 			},
 		},
-		// Now test where the server flip flops between "2.0;snappy" and "0.1.0" only.
+		// Opposite, v1 talking to v2 only server.
 		{
-			success: false, name: "flip flopping", qmRwFormat: config.RemoteWriteProtoMsgV2, rwFormat: Version2, steps: []contentNegotiationStep{
-				{lastRWHeader: "2.0;snappy", compression: "snappy", behaviour: &ErrRenegotiate{"", 406}, attemptString: "0,1,snappy,HTTP 406: msg: "},
-				{lastRWHeader: "0.1.0", compression: "snappy", behaviour: &ErrRenegotiate{"", 406}, attemptString: "0,0,snappy,HTTP 406: msg: "},
-				{lastRWHeader: "2.0;snappy", compression: "snappy", behaviour: &ErrRenegotiate{"", 406}, attemptString: "0,1,snappy,HTTP 406: msg: "},
-				// There's no 4th attempt as we do a maximum of 3 sending attempts (not counting retries).
+			success: false, name: "v1 talks to v2 that gives 400 or 415", senderProtoMsg: config.RemoteWriteProtoMsgV1, receiverProtoMsg: config.RemoteWriteProtoMsgV2, steps: []contentNegotiationStep{
+				{behaviour: errors.New("pretend unrecoverable err"), attemptString: "0,pretend unrecoverable err"},
 			},
 		},
-	}
-
-	queueConfig := config.DefaultQueueConfig
-	queueConfig.BatchSendDeadline = model.Duration(100 * time.Millisecond)
-	queueConfig.MaxShards = 1
-
-	// We need to set URL's so that metric creation doesn't panic.
-	writeConfig := baseRemoteWriteConfig("http://test-storage.com")
-	writeConfig.QueueConfig = queueConfig
-
-	conf := &config.Config{
-		GlobalConfig: config.DefaultGlobalConfig,
-		RemoteWriteConfigs: []*config.RemoteWriteConfig{
-			writeConfig,
+		{
+			success: false, name: "v1 talks to (broken) v2 that tries to unmarshal v1 payload with v2 proto", senderProtoMsg: config.RemoteWriteProtoMsgV1, receiverProtoMsg: config.RemoteWriteProtoMsgV2, steps: []contentNegotiationStep{
+				{behaviour: nil, attemptString: "0,ok"}, // It's ok, because payload can be unmarshaled just fine, but it's empty.
```

Comment:
ACAIFT `attemptString` is not used anywhere. After these changes, only the `behaviour` fields from the `steps: []contentNegotiationStep` are used, and only tell the TestClient to return a sequence of errors. I would replace the `steps : []contentNegotiationStep` with `errsSeq: []error` or similar

Reply:
hah lol, you are right. Thanks! will fix